semantic/io.cpp

```diff
@@ -64,7 +64,18 @@ void parse(math::Point3 &point, const Json::Value &value)
     Json::unpack(value, "Point3", point(0), point(1), point(2));
 }
 
-void parse(math::Points3 &points, const Json::Value &value)
+void parse(std::string& str, const Json::Value &value)
```

Comment:
Formatting inconsistency of ampersands; I personally prefer to emphasize it's a modifier of a type and write them adjacent to the type. E.g. `void parse(std::string& str, const Json::Value& value)`.

Reply:
And I exactly prefer it the other way around :man_shrugging: 
Technically, it modifes the variable, not the type.

Reply:
It’s a matter of preference; but consistency is the key. It’s quite weird to combine both styles on a single line. Btw Bjarne Stroustrup, Scott Meyers, Herb Sutter, et al. write it adjacent to the type.

Reply:
This is again a matter of taste, in fact I am used to center alignment... In our clang-format we defined left alignment. Choose whatever, I only suggest to be consistent - at least within a line.

Reply:
Fixed